subprojects/resources-s3/src/integTest/groovy/org/gradle/integtests/resource/s3/maven/MavenPublishS3IntegrationTest.groovy

```diff
@@ -77,4 +77,56 @@ publishing {
         module.assertPublishedAsJavaModule()
         module.parsedPom.scopes.isEmpty()
     }
+
+  
+    def "can publish to a S3 Maven repository with IAM"() {
+        given:
+        System.setProperty('aws.accessKeyId', 'someKey')
```

Comment:
I'd think that the system properties would have to be provided to the `executor` instance.